src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeDescriptor-Button-Actions.SummaryAdmin.cshtml

```diff
@@ -1,2 +1,2 @@
 @model ShapeViewModel<ShortcodeDescriptor>
-<button class="btn btn-primary btn-sm" v-on:click="insertShortcode('@Model.Value.DefaultValue')">@T["Add"]</button>
+<button class="btn btn-primary btn-sm" v-on:click="insertShortcode('@Model.Value.DefaultValue.Replace("'", @"\'")')">@T["Add"]</button>
```

Comment:
Use `JavascriptEncoder.Encode` instead of replacing the single quote. It will handle all values.
another better way is to render it in an attribute, and read the attribute in javascript, this way Razor will encode it by itself. We do it in many views already to render urls.

Reply:
Done